### DISCUSSION TITLE
Admob Open Bidding not working as expected

### DISCUSSION THREAD

**Discussion by tarouboy** [invertase/react-native-firebase#5234] [Q&A]
I have joined the Admob Open Bidding beta and tried to add the Facebook Audience Network into the mediation.
After checking the guide line I couldn't find any further steps are required to enable it. I am not sure if I've missed something.

From the admob [instruction](https://developers.google.com/admob/android/mediate), I can only see the SDK have to be above 18.3.0+ for Android and 7.53.1 for iOS.

My Podfile:
`pod 'Google-Mobile-Ads-SDK', '7.69.0'`

My build.gradle:
```
implementation 'com.google.ads.mediation:facebook:6.3.0.1'
implementation 'androidx.annotation:annotation:1.0.0'
implementation 'com.facebook.android:audience-network-sdk:6.+'
implementation 'com.google.android.gms:play-services-ads:20.0.0'
```

However, from the Admob mediation report, I can see requests are being passed to FAN with 0% match rate ( which means failed ).
<img width="961" alt="Screenshot 2021-04-30 at 12 22 06 AM" src="https://user-images.githubusercontent.com/21119498/116584631-24806380-a94a-11eb-9100-5c49defd96ef.png">

At the same, Facebook announced that [we must use Open Bidding after September](https://www.facebook.com/audiencenetwork/news-and-insights/audience-network-to-become-bidding-only-beginning-with-ios-in-2021/).

So is it because the current version of rn-firebase/admob or I have missed something on integration?
Any kind of help is much appreciated.

**Comment by mikehardy** [comment]
> implementation 'com.google.android.gms:play-services-ads:20.0.0'

This looks dangerous right now in this repo, have you checked the pinned issue about admob? #5150

Other than that, I'm not sure. It sounds like what you must do is verify that the correct versions are installed while using react-native-firebase, if that's possible.

You did not include any of the react-native-firebase versions, so I'm not sure what you're running but I suggest:

- make sure you are on the recently-released 11.4.1 so you are current
- run `cd android && ./gradlew :app:dependencies > dependencies.txt` and search it for the dependencies you expect, verify their version
- do the same with `ios/Podfile.lock` to make sure you have the dependencies you expect

If they are the correct version, there is nothing more to do in this module, you have something going on with your integration or configuration.

If they are on the wrong version, you might check the output of the same items from a clean / current demo script run https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh to see if they will work.

If the demo script produces version output that is equal or greater to minimum requirements then there is something else going on with your project + dependencies that is altering installed versions, and it won't result in a change here.

If the demo script *does not produce compatible versions* then it may be due to the linked issue above, and it's pending work here until we port to the new v27 release of firebase-android-sdk

I'm curious to know the findings, please report back

**Reply by tarouboy** [reply]
Thank you! It helps a lot 🙏
Im on react-native-firebase 11.3.3

**Comment by tarouboy** [comment]
I have double checked and quite sure the both the Admob and Facebook Audience Network Adaptor version matched the min req for open-bidding.
However, ad requests are passed but still 0% match rate from Admob mediation reports.

I have found the below parts from this admob instruction.
I think the current method used to initialize the Admob SDK is outdated:
https://developers.google.com/ad-manager/mobile-ads-sdk/ios/early-access/open-bidding

> If ads are loaded before an open bidding adapter has finished initializing, that adapter won't participate in the ad request.
> 
> The initialization completionHandler is called once all open bidding adapters have been initialized, or after a 10 second timeout, whichever comes first. If your first ad request is time sensitive, you can always check the initializationStatus property on GADMobileAds at any time to verify which adapters have been initialized.

```
@import GoogleMobileAds;

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application
    didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {

  GADMobileAds *ads = [GADMobileAds sharedInstance];
  [ads startWithCompletionHandler:^(GADInitializationStatus *status) {
    // Optional: Log each adapter's initialization latency.
    NSDictionary *adapterStatuses = [status adapterStatusesByClassName];
    for (NSString *adapter in adapterStatuses) {
      GADAdapterStatus *adapterStatus = adapterStatuses[adapter];
      NSLog(@"Adapter Name: %@, Description: %@, Latency: %f", adapter,
            adapterStatus.description, adapterStatus.latency);
    }

    // Start loading ads here...
  }];
  return YES;
}

@end
```

Not sure how to update the code tho...

**Comment by manuhook** [comment]
It works on our app but open bidding is not really performant with Facebook. Our match rate is 0,01 % with 20 millions requests. 
On iOS they announced to stop the waterfall system on March 31, 2021 but it's still working, and the match rate is much better (50%), did you try the setup with Waterfall ?

**Reply by tarouboy** [reply]
Waterfall is working perfectly, just trying to make it work with open bidding. 
Seems like it will be the new trend and going to replace waterfall.

I think 0.01% means it's not passing through. Have you checked Facebook's reports? If it's working the no. of requests should be very close to the total number of admob requests.
